api/capi/include/platform/ml-api-common.h

```diff
@@ -0,0 +1,46 @@
+/* SPDX-License-Identifier: LGPL-2.1-only */
+/**
+ * NNStreamer API / Tizen Machine-Learning API Common Header
+ * Copyright (C) 2020 MyungJoo Ham <myungjoo.ham@samsung.com>
+ */
+/**
+ * @file	ml-api-common.h
+ * @date	07 May 2020
+ * @brief	Dummy ML-API Common Header from nnstreamer
+ * @see		https://github.com/nnstreamer/nnstreamer
+ * @author	MyungJoo Ham <myungjoo.ham@samsung.com>
+ * @bug		No known bugs except for NYI items
+ *
+ * @details
+ *      More entries might be migrated from nnstreamer.h if
+ *    other modules of Tizen-ML APIs use common data structures.
+ */
+#ifndef __ML_API_COMMON_H__
+#define __ML_API_COMMON_H__
+
+#include <tizen_error.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif /* __cplusplus */
+
+/**
+ * @brief Enumeration for the error codes of NNStreamer.
+ * @since_tizen 5.5
+ */
+typedef enum {
+  ML_ERROR_NONE                 = TIZEN_ERROR_NONE, /**< Success! */
+  ML_ERROR_INVALID_PARAMETER    = TIZEN_ERROR_INVALID_PARAMETER, /**< Invalid parameter */
+  // ML_ERROR_STREAMS_PIPE         = TIZEN_ERROR_STREAMS_PIPE, /**< Cannot create or access the pipeline. not used in nntrainer */ 
+  // ML_ERROR_TRY_AGAIN            = TIZEN_ERROR_TRY_AGAIN, /**< The pipeline is not ready, yet (not negotiated, yet) not used in nntrainer */
+  ML_ERROR_UNKNOWN              = TIZEN_ERROR_UNKNOWN,  /**< Unknown error */
+  ML_ERROR_TIMED_OUT            = TIZEN_ERROR_TIMED_OUT,  /**< Time out */
+  ML_ERROR_NOT_SUPPORTED        = TIZEN_ERROR_NOT_SUPPORTED, /**< The feature is not supported */
+  ML_ERROR_PERMISSION_DENIED    = TIZEN_ERROR_PERMISSION_DENIED, /**< Permission denied */
+  ML_ERROR_OUT_OF_MEMORY        = TIZEN_ERROR_OUT_OF_MEMORY, /**< Out of memory (Since 6.0) */
+} ml_error_e;
```

Comment:
1. Add defensive mechanism:
E.g.,
```
#ifdef __TIZEN__
#error TIZEN build should not use this
#endif
```

2. Define error code directly
e.g.,
```
#define ML_ERROR_OUT_OF_MEMORY    (ENOMEM)
```

Reply:
You don't need to have a separated file tizen_error.h as long as you have this file for non-Tizen platforms. Merge them (by the comment above)

Reply:
Thank you for the review. It is fixed :)

Reply:
"2." is not reflected.

Reply:
Uhh I misunderstood "2." Fixed!